### ISSUE TITLE
Add `name` configuration property under `overrides`

### ISSUE THREAD

**Issue by ghost** [stylelint/stylelint#7854]
### What is the problem you're trying to solve?

ESLint recently allowed users to write a "name" option in eslint.config.js. Reference: <https://github.com/eslint/eslint/commit/4d11e567baff575146fd267b3765ab2c788aa1e5>

Use case 1: to describe the reasons for enabling or disabling some rules

```js
import { defaultEslintRules } from '@eslint/js';

export default [
  defaultEslintRules.configs.recommended,
  {
    files: ['file1.js'],
    name: "In this case, the max-lines rule was disabled temporarily to avoid an edge-case scenario, and we plan to enable back in the future.",
    rules: {
      'max-lines': 'off',
    },
  },
  {
    files: ['file2.js'],
    name: "In this case, max-lines rule was disabled to follow our company guideline.",
    rules: {
      'max-lines': 'off',
    },
  },
  {
    files: ['file3.js'],
    name: "Rules disabled by Maria, should remain like this. Please don't touch.",
    rules: {
      'max-lines': 'off',
      'max-lines-per-function': 'off',
      'max-statements': 'off',
      'no-magic-numbers': 'off',
    },
  },
];
```

Use case 2: to name the group of rules.

```js
name: 'eslint/recommended'
```

or

```js
name: 'eslint/strict'
```

### What solution would you like to see?

I would like to write a "name" option in stylelint.config.js too.

For example:
```js
extends: [defaultConfig],
overrides: [
  {
    files: ['src/component/star.css'],
    name: 'our-rules/strict',
    rules: {
      'property-allowed-list': null,
      'property-disallowed-list': null,
    },
  },
],
```

or

```js
extends: [defaultConfig],
overrides: [
  {
    files: ['src/component/star.css'],
    name: 'These rules are disabled because of ...',
    rules: {
      'property-allowed-list': null,
      'property-disallowed-list': null,
    },
  },
],
```

**Comment by Mouvedia** [comment]
> [The name is used in error messages and the config inspector to help identify which configuration object is being used.](https://eslint.org/docs/head/use/configure/configuration-files#configuration-naming-conventions)

**Comment by ybiquitous** [comment]
@mrredtree Thanks for the suggestion. I'm unsure about the `name` option, so can you explain use cases?

**Comment by Mouvedia** [comment]
I think it would be useful if the config contains something that triggers a deprecation warning.
e.g.

```json
{
  "extends": [
    "foo",
  ],
  "overrides": [
    {
      "name": "foo"
    },
    {
      "name": "bar"
    },
  ]
}
```

Since both configs are in the same file, the `name` permits to differentiate them when debugging based on the details returned by `process.emitWarning`.
An alternative would be to simply return a stack trace using [the `ctor` argument of `emitWarning`](https://nodejs.org/api/process.html#processemitwarningwarning-type-code-ctor).

**Comment by ybiquitous** [comment]
@mrredtree Thanks for the explanation. The new option sounds good to me. And adding it would not be problematic: just allowing `name` in a config.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.
